utils/setup.php

```diff
@@ -221,9 +252,6 @@
 if ($aCMDResult['create-functions'] || $aCMDResult['all']) {
     info('Create Functions');
     $bDidSomething = true;
-    if (!file_exists(CONST_InstallPath.'/module/nominatim.so')) {
-        fail('nominatim module not built');
-    }
```

Comment:
In the case where `setup.php` is running on a host other than the database server (which is likely to be the normal case when the application baseline is on a web server host and the database server is on a separate host), the `nominatim.so` file will not and should not exist on the local filesystem.

I was thinking that this check in the `--setup-db` section would be a sufficient notification that the module is absent on the database server (whether remote or local):
```    // Try accessing the C module, so we know early if something is wrong
    // and can simply error out.
    $sSQL = "CREATE FUNCTION nominatim_test_import_func(text) RETURNS text AS '";
    $sSQL .= $modulePath."/nominatim.so', 'transliteration' LANGUAGE c IMMUTABLE STRICT";
    $sSQL .= ';DROP FUNCTION nominatim_test_import_func(text);';
    $oResult = $oDB->query($sSQL);
```

Reply:
`--create-function` is rerun occasionally without `--setup-db`, so I think repeating the check makes sense here. How about you move the check from `--setup-db` in a helper function and then call that function again here. I wouldn't want to remove the check from `--setup-db` either because you wouldn't want to do the full osm2pgsql import before realising that something is wrong with the nominatim.so.

Reply:
I refactored it as you described.